vertica_python/vertica/connection.py

```diff
@@ -171,38 +176,39 @@ def _append(self, host, port):
             self._logger.error(err_msg)
             raise ValueError(err_msg)
 
-        self.address_deque.append((host, port, False))
+        self.address_deque.append(_AddressEntry(resolved=False, data=(host, port)))
 
     def push(self, host, port):
-        self.address_deque.appendleft((host, port, False))
+        self.address_deque.appendleft(_AddressEntry(resolved=False, data=(host, port)))
 
     def pop(self):
         self.address_deque.popleft()
 
     def peek(self):
-        # do lazy DNS resolution, return the leftmost DNS-resolved address
+        # do lazy DNS resolution, returning the leftmost socket.getaddrinfo result
         if len(self.address_deque) == 0:
             return None
 
         while len(self.address_deque) > 0:
-            host, port, is_dns_resolved = self.address_deque[0]
-            if is_dns_resolved:
-                # return a resolved address
+            entry = self.address_deque[0]
+            if entry.resolved:
+                # return a resolved sockaddrinfo
                 self._logger.debug('Peek at address list: {0}'.format(list(self.address_deque)))
-                return (host, port)
+                return entry.data
             else:
                 # DNS resolve a single host name to multiple IP addresses
                 self.address_deque.popleft()
+                host, port = entry.data
                 try:
-                    resolved_hosts = socket.getaddrinfo(host, port, socket.AF_INET, socket.SOCK_STREAM)
+                    resolved_hosts = socket.getaddrinfo(host, port, type=socket.SOCK_STREAM)
```

Comment:
Regression on Python 2.7. https://docs.python.org/2/library/socket.html#socket.getaddrinfo
```suggestion
                    resolved_hosts = socket.getaddrinfo(host, port, 0, socket.SOCK_STREAM)
```